xonsh/prompt/vc.py

```diff
@@ -175,19 +164,27 @@ def current_branch():
 
 
 def _git_dirty_working_directory(q, include_untracked):
-    status = None
     denv = builtins.__xonsh__.env.detype()
     try:
-        cmd = ["git", "status", "--porcelain"]
+        # Borrowing this conversation
+        # https://github.com/sindresorhus/pure/issues/115
         if include_untracked:
-            cmd.append("--untracked-files=normal")
+            cmd = [
+                "git",
+                "status",
+                "--porcelain",
+                "--quiet",
+                "--untracked-files=normal",
+            ]
         else:
-            cmd.append("--untracked-files=no")
-        status = subprocess.check_output(cmd, stderr=subprocess.DEVNULL, env=denv)
+            cmd = ["git", "diff", "--no-ext-diff", "--quiet"]
+        child = subprocess.Popen(cmd, stderr=subprocess.DEVNULL, env=denv)
+        child.communicate()
```

Comment:
please just do a subprocess.run() or check_output(), rather than making a Popen object and using communicate